File: src/ERP/acme.atena.domain/DTO/Util/Endereco.cs
using acme.atena.domain.DTO.Person;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace acme.atena.domain.DTO.Util
{
    public class Endereco: AbstractEntity
    {
        public Endereco()
        {
            EnderecoClientes = new HashSet<EnderecoCliente>();
            EnderecoFornecedores = new HashSet<EnderecoFornecedor>();
            EnderecoUsuarios = new HashSet<EnderecoUsuario>();

        }
        public string Cep { get; set; }
        public string Pais { get; set; }
        public string Estado { get; set; }
        public string Cidade { get; set; }
        public string Bairro { get; set; }
        public string Rua { get; set; }

        public virtual ICollection<EnderecoCliente> EnderecoClientes { get; set; }
        public virtual ICollection<EnderecoUsuario> EnderecoUsuarios { get; set; }
        public virtual ICollection<EnderecoFornecedor> EnderecoFornecedores { get; set; }
        public virtual ICollection<EnderecoEmpresa> EnderecoEmpresas { get; set; } = new HashSet<EnderecoEmpresa>();

    }
}

File: site/coreui-free-angular-admin-template-master/src/app/view-model/response/util/endereco.ts
import { EnderecoCliente } from './endereco-cliente';
import { EnderecoUsuario } from './endereco-usuario';
import { AbstractEntity } from "../abstract-entity";
import { EnderecoFornecedor } from './endereco-fornecedor';
import { EnderecoEmpresa } from './endereco-empresa';

export class Endereco extends AbstractEntity{
   cep : string;
   pais: string;
   estado: string;
   cidade: string;
   bairro: string;
   rua: string;

  enderecoUsuarios : EnderecoUsuario[]= [];
  enderecoClientes : EnderecoCliente[]= [];
  enderecoFornecedores : EnderecoFornecedor[]= [];
  enderecoEmpresas: EnderecoEmpresa[] = [];
}
